wordpress-mobile/WordPress-Android
# Follow recommended blog from "recommended_blogs" discovery card

Task: #12834

This PR introduces `follow` action from `recommended_blogs` card in the new Discover tab. 
I've tried to reuse as much `follow` logic as possible. I had to slightly modify the use-case to make it more versatile and not tightly coupled with the `Post` object.  

The first implementation was storing all the information about the recommended blogs in the `tbl_discover_cards` table. After discussing it with @ashiagr we came to the conclusion that it makes more sense to reuse `tbl_recommended_blogs` and store the `ReaderBlog` object there, and only store simplified JSON in the `tbl_recommended_blogs`. This is how it was done for the `Post` card, so following the same pattern makes it easier to follow. This commit https://github.com/wordpress-mobile/WordPress-Android/pull/12960/commits/731b18803623ae0c613a111e7998f744c79a98ca contains all the changes that were needed to do that.

Questions:
1. Should we show "enable notification" snack when following the Site from the "recommended_blogs" card?

| Light  | Dark  |
|---|---|
| ![follow_site_light](https://user-images.githubusercontent.com/12660801/93262740-0728cb80-f7a5-11ea-882d-65d61933570b.gif)  |  ![follow_site_dark](https://user-images.githubusercontent.com/12660801/93262808-2293d680-f7a5-11ea-97a1-fd83a1386063.gif) |
| ![Screenshot_20200924-201815](https://user-images.githubusercontent.com/12660801/94184084-ad5e8a80-fea3-11ea-9712-97148f972f12.jpg) | ![Screenshot_20200924-201744](https://user-images.githubusercontent.com/12660801/94184132-bc453d00-fea3-11ea-8f0a-c76c136eb9a8.jpg) |
| ![Screenshot_20200924-201834](https://user-images.githubusercontent.com/12660801/94184431-2eb61d00-fea4-11ea-9949-9c1e50618d88.jpg) | ![Screenshot_20200924-201856](https://user-images.githubusercontent.com/12660801/94184486-4392b080-fea4-11ea-82fc-287a4246f069.jpg) |


To test:
- open Discovery tab
- scroll to 'Sites to follow' card
- click follow/unfollow IconButton
- verify site was followed/unfollowed


PR submission checklist:
